chalicelib/app_utils.py

```diff
@@ -38,6 +38,8 @@
 )
 from .s3_connection import S3Connection
 
+FAVICON_4DN = 'https://data.4dnucleome.org/static/img/favicon-fs.ico'
```

Comment:
This is not right. If you are using the "fourfront-cgap" environment, it would continue to get the favicon from Fourfront. The only benefit you get from defining it here (instead of the template) is that you can programmatically change it if you wanted.

Instead, you should pass the current Fourfront/CGAP server (i.e. `connection.ff_server` to the template as a new variable and then render `{{server + '/static/img/favicon-fs.ico'}}` wherever you need the favicon. Or you could add the favicon path to the variable before you pass it to the template; same deal.

If you have multiple environments, that's where it gets tricky. Currently, `view_foursight` is the only templating function that supports multiple environments, in which case I think you can just choose one randomly to get the favicon from. Or use `FAVICON_4DN` as a default. Up to you.

Reply:
Is it fair to assume the asset will always be available regardless of what `server` is? How can I best detect if isn't prior to rendering the page? I'd need to know to fallback to `FAVICON_4DN`, which I'll probably rename to `DEFAULT_FAVICON`, before rendering the page, no? I guess I could hardcode the default into the html and if the favicon 404's use that one instead?

My thought for multiple environments would be to just pick the first one and use that.